### PR TITLE
Soback 22 url search implemented

### PR DESCRIPTION
**Pull request by jesperlauridsen**
So, this is the stab at this task (feedback very welcome)

When URL search is chosen, "URL" is shown, to indicate we want an URL (if the query starts with url_norm), we assume people know what they're doing, and remove "URL".

Unless the query starts with http:// or https:// (or url_norm:), it'll turn red. If its okey, it'll turn our normal color
If you still try to search while it's red, an error will show.

When you've searched, we normalize the URL, fire off the search, keep the tick, and apply the "url_norm:"" infront of the query, and " after it.

Now it becomes tricky. We're still in URL search land, because we want to make sure if people alter their query, that we normalize it. This means, that if a URL search is fired again, we check if "url_norm" is in the beginning, and if it is, we filter that away, and  the " at the end, and submit that string to the normalizer, and fire off a search with what we get back. 

After you've search, we show you the original query on top of the search-field, to let you see how your query was normalized.

I don't think this is by any means the best solution, but it's what I've been able to come up with so far. Any suggestions to how it could be better would be greatly appreciated. 

If we do a full rewrite, perhaps it should be a modal that opens when you want a URL search - to make sure it's a one-time-thing - but this would then limit the users ability to quickly change the URL and fire off a new search if a typo was made, or for whatever reason.

### CODE REVIEW

**Review by jorntx — COMMENTED** [review]
I like the solution with the hint in the search box very much👍  

I have a few suggestions for code improvement but overall it looks fine. As a side note I can see that SearchBox.vue is ballooning and when the last few search features a implemented we should look at a refactor of that component.

Functionality wise:
- I get a backend error saying "Url not starting with http:// or https://" when I search for https urls. The request param is correct.
- Consider clearing all current facets and search state when user checks the url search check box or at least reset current search state when first url search is submitted.

File: src/js/src/components/SearchBox.vue [hunk 1]
```diff
@@ -100,6 +108,29 @@ export default {
            this.$_pushSearchHistory('SolrWayback', this.query, this.searchAppliedFacets, this.solrSettings)
         }
         else if(this.solrSettings.urlSearch) {
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/js/src/components/SearchBox.vue:108]: I'll keep that in mind for the refactor.

File: src/js/src/views/SolrWayback.vue [hunk 2]
```diff
@@ -39,29 +39,36 @@ export default {
...
       this.requestImageSearch({query:this.query})
-      return
     }
     else if(this.solrSettings.urlSearch) {
-      return
+      let queryString = ''
+      if(this.query.substring(0,10) === 'url_norm:"') {
+        queryString = this.query.replace('url_norm:"', '')
+        queryString.substring(queryString.length-1, queryString.length) === '"' ? queryString.slice(0,-1) : null
+        this.updateQuery(queryString)
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/js/src/views/SolrWayback.vue:81]: You've seen this 3 times - all 3 times are part of the imageSearch function. I won't touch it now, but again, it'll all be in a mixin for one searchfunction very soon!

File: src/js/src/store/modules/search.store.js [hunk 3]
```diff
@@ -22,9 +22,7 @@ const actions = {
   setLoadingStatus( {commit}, param) {
     commit('setLoadingStatus', param)
   },
-  updateQuery ( {commit, rootState}, param, foo, bar) {
-    console.log(rootState.Notifier)
-    console.log(bar)
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/js/src/store/modules/search.store.js:27]: :+1:

File: src/js/src/store/modules/notifier.store.js [hunk 4]
```diff
@@ -29,6 +29,7 @@
 
     dismissNotification(state, notification) {
       Vue.delete(state.notifications, notification.__ob__.vmCount)
+      this.dispatch('Search/setLoadingStatus', false)
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/js/src/store/modules/notifier.store.js:32]: :+1:

File: src/js/src/components/SearchBox.vue [hunk 5]
```diff
@@ -169,8 +226,19 @@ export default {
...
+      this.futureUrlSearch = false
+      this.futureImgSearch = false
       this.resetSearchState()
     },
+    validateUrl(testString) {
+      return testString.substring(0,7) === 'http://' || 
+             testString.substring(0,8) === 'https://' || 
+             testString.substring(0,10) === 'url_norm:"'
+      ? true
+      : false
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/js/src/components/SearchBox.vue]: You're very good at spotting these. I don't know why I'm so determined to spell it out each time o_O

File: src/js/src/components/SearchBox.vue [hunk 6]
```diff
@@ -130,24 +166,45 @@ export default {
...
+            this.$_pushSearchHistory('SolrWayback', queryString, this.searchAppliedFacets, this.solrSettings)
+          }
+          else {
+            this.setNotification({
+          	title: 'We are so sorry!',
+            text: 'This query is not valid. the url must start with \'http://\' or \'https://\'',
+            type: 'error',
+            timeout: false
+          })
+          }
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/js/src/components/SearchBox.vue:200]: You're absolutely right. I won't touch it here, but this is a part of the refactor of the search function im planning. In mounted and in handleSubmit, much of the same code is used to check what kind of search should be triggered. I'm gonna fix all this in SOBACK-30.

File: src/js/src/components/SearchBox.vue [hunk 7]
```diff
@@ -100,6 +108,29 @@ export default {
...
+            this.$_pushSearchHistory('SolrWayback', this.query, this.searchAppliedFacets, this.solrSettings)
+          }
+          else {
+            this.setNotification({
+          	title: 'We are so sorry!',
+            text: 'This query is not valid. the url must start with \'http://\' or \'https://\'',
+            type: 'error',
+            timeout: false
+          })
+          }
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/js/src/components/SearchBox.vue:131]: You're absolutely right. I won't touch it here, but this is a part of the refactor of the search function im planning. In mounted and in handleSubmit, much of the same code is used to check what kind of search should be triggered. I'm gonna fix all this in SOBACK-30.

File: src/js/src/components/SearchBox.vue [hunk 8]
```diff
@@ -1,13 +1,20 @@
...
              type="text"
              autofocus
-             class=""
-             autocomplete="off"
-             placeholder="Enter search term">
+             :class="futureUrlSearch 
+               ? validateUrl(futureQuery) === false 
+                 ? futureQuery.substring(0,8) === 'url_norm' ? 'urlNotTrue' : 'urlNotTrue urlSearchActivated'
+                 : futureQuery.substring(0,8) === 'url_norm' ? '' : 'urlSearchActivated' 
+               : ''"
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/js/src/components/SearchBox.vue]: Yea, good call. Makes it easier to read.

File: src/js/src/components/SearchBox.vue (hunk 7 above)
[the remark this answers is not in the document]
Comment (jorntx) [reply src/js/src/components/SearchBox.vue:131]: 👍

File: src/js/src/views/SolrWayback.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (jorntx) [reply src/js/src/views/SolrWayback.vue:81]: 👍

**Pull request merged**